src/node_report.cc

```diff
@@ -0,0 +1,790 @@
+
+  fprintf(fp, "\n================================================================================");
+  fprintf(fp, "\n==== JavaScript Heap and Garbage Collector =====================================\n");
+  HeapSpaceStatistics v8_heap_space_stats;
+  // Loop through heap spaces
+  for (size_t i = 0; i < isolate->NumberOfHeapSpaces(); i++) {
+    isolate->GetHeapSpaceStatistics(&v8_heap_space_stats, i);
+    fprintf(fp, "\nHeap space name: %s", v8_heap_space_stats.space_name());
+    fprintf(fp, "\n    Memory size: ");
+    WriteInteger(fp, v8_heap_space_stats.space_size());
+    fprintf(fp, " bytes, committed memory: ");
+    WriteInteger(fp, v8_heap_space_stats.physical_space_size());
+    fprintf(fp, " bytes\n    Capacity: ");
+    WriteInteger(fp, v8_heap_space_stats.space_used_size() +
+                           v8_heap_space_stats.space_available_size());
+    fprintf(fp, " bytes, used: ");
+    WriteInteger(fp, v8_heap_space_stats.space_used_size());
+    fprintf(fp, " bytes, available: ");
+    WriteInteger(fp, v8_heap_space_stats.space_available_size());
+    fprintf(fp, " bytes");
+  }
+
+  fprintf(fp, "\n\nTotal heap memory size: ");
+  WriteInteger(fp, v8_heap_stats.total_heap_size());
+  fprintf(fp, " bytes\nTotal heap committed memory: ");
+  WriteInteger(fp, v8_heap_stats.total_physical_size());
+  fprintf(fp, " bytes\nTotal used heap memory: ");
+  WriteInteger(fp, v8_heap_stats.used_heap_size());
+  fprintf(fp, " bytes\nTotal available heap memory: ");
+  WriteInteger(fp, v8_heap_stats.total_available_size());
+  fprintf(fp, " bytes\n\nHeap memory limit: ");
+  WriteInteger(fp, v8_heap_stats.heap_size_limit());
+  fprintf(fp, "\n");
+}
+
+#ifndef _WIN32
+/*******************************************************************************
+ * Function to print resource usage (Linux/OSX only).
+ *
+ ******************************************************************************/
+static void PrintResourceUsage(FILE *fp) {
+  fprintf(fp, "\n================================================================================");
+  fprintf(fp, "\n==== Resource Usage ============================================================\n");
+
+  // Process and current thread usage statistics
+  struct rusage stats;
+  fprintf(fp, "\nProcess total resource usage:");
+  if (getrusage(RUSAGE_SELF, &stats) == 0) {
+    fprintf(fp,"\n  User mode CPU: %ld.%06ld secs", stats.ru_utime.tv_sec, stats.ru_utime.tv_usec);
+    fprintf(fp,"\n  Kernel mode CPU: %ld.%06ld secs", stats.ru_stime.tv_sec, stats.ru_stime.tv_usec);
+    fprintf(fp,"\n  Maximum resident set size: ");
+    WriteInteger(fp,stats.ru_maxrss * 1024);
+    fprintf(fp," bytes\n  Page faults: %ld (I/O required) %ld (no I/O required)", stats.ru_majflt, stats.ru_minflt);
+    fprintf(fp,"\n  Filesystem activity: %ld reads %ld writes", stats.ru_inblock, stats.ru_oublock);
+  }
+
+  fprintf(fp, "\n\nEvent loop thread resource usage:");
+  if (getrusage(RUSAGE_THREAD, &stats) == 0) {
```

Comment:
Only some platforms have RUSAGE_THREAD.  Try wrapping it in an `#ifdef RUSAGE_THREAD`.

I'm not 100% sure if it's guaranteed to be a define, POSIX only requires that it's a "symbolic constant", but on linux and freebsd it's a define (and those are the only two that have RUSAGE_THREAD, I think.  OS X doesn't have it but you can query it through proc_info(), I think.)